main.js

```diff
@@ -0,0 +1,152 @@
+const list = ['Apple', 'Bread', 'Carrot', 'Dumplings',
+             'Eggs','Fish','Garlic','Honey','Ice cream','Jam'];
+const nodes = list.map((item) => `<span>${item}</span>`);
+
+let initialState = {
+  items: list,
+  nodes: nodes,
+  up: true,
+  down: true
+}
+
+function reducer(state, action) {
+  let up = false;
+  let down = false;
+  
+  switch (action.type) {
+    case 'select':
+        if (action.index === 0) {
+          up = true;
+        } else if (action.index === state.items.length-1) {
+          down = true;
+        }
+      return {
+        ...state,
+        nodes: state.items.map((item, index) => {
+          if (index === action.index) {
+            item = `<span class='active'>${item}</span>`;
+          } else {
+            item = `<span>${item}</span>`;
+          }
+          return item
+        }),
+        up: up,
+        down: down
+      }
+    case 'unselect': 
+      return {
+        ...state,
+        nodes: state.items.map(item => `<span>${item}</span>`),
+        up: true,
+        down: true
+      }
+    case 'moveUp':
+      return move(state, action);
+    case 'moveDown':
+      return move(state, action);
+    default:
+      return state;
+  } 
+}
+
+function move(state, action) {
+  const newItems = [...state.items];
+  const itemToMove = newItems.splice(action.index, 1);
+  const newNodes = [...state.nodes];
+  const active = newNodes.splice(action.index, 1);
+  let up;
+  let down;
+
+  if (action.index === 1) {
+    up = true;
+  } else if (action.index === state.items.length-2) {
```

Comment:
why `- 2`?
```suggestion
  } else if (action.index === state.items.length - 2) {
```

Reply:
-2 because when i am moving elem with index = state.items.length - 2 i need  assign to the  button called "move down" attribute disabled = true.